smarts/sstudio/types.py

```diff
@@ -525,13 +525,12 @@ def pick_remaining_shape_after_split(geometry_collection, length, lane):
             if len(lane_shape) == 1:
                 return lane_shape[0]
 
-            expected_bbox_area = lane.getWidth() * length
+            # We assume that there are only two splited shapes to choose from
             keep_index = 0
-            if (lane_shape[0].minimum_rotated_rectangle.area - expected_bbox_area) < (
-                lane_shape[1].minimum_rotated_rectangle.area - expected_bbox_area
-            ):
+            if lane_shape[1].minimum_rotated_rectangle.contains(expected_point):
```

Comment:
The core of this change is to use the midpoint inside the cut geometry to check if the second cut is viable.